torch/_subclasses/fake_tensor.py

```diff
@@ -234,28 +233,14 @@ def invalidate_constant_aliases(self, tensor):
         del self.constant_storage_mapping[weak_st]
 
     def _get_memo(self, t):
-        if WeakIdRef(t) in self.tensor_memo:
-            out = self.tensor_memo[WeakIdRef(t)]
-            out._fix_weakref()
-            return out
-        return None
+        tid = self.meta_converter.describer.lookup_tensor.get(t)
+        if tid is None:
+            return None
+        return self.tensor_memo.get(tid)
 
     def set_tensor_memo(self, t, v):
-        th = WeakIdRef(t)
-
-        # hold a weak ref to self, otherwise it will be kept alive
-        # by the del_ten closure
-        self_weak_ref = weakref.ref(self)
-
-        def del_ten():
-            self_ref = self_weak_ref()
-            if self_ref is None:
-                return
-            # on shutdown, th may not be in memo
-            self_ref.tensor_memo.pop(th, None)
-
-        weakref.finalize(t, del_ten)
-        self.tensor_memo[th] = v
+        tid = self.meta_converter.describer.get_tensor_id(t)
+        self.meta_converter.tensor_memo[tid] = v
```

Comment:
We're not deleting the cache entry anymore when the tensor dies anymore, but I guess that's okay since its cheap to keep a bunch of FakeTensor alive?

Reply:
Yes. So specifically, previously we would have cleaned up the table entry when either the real tensor OR the fake tensor died. Now, I will maintain one table entry per real/fake tensor (separately). I don't think this is a big deal.

Reply:
In the case where someone runs :

```
x = torch.rand([4])
mode = FakeTensorMode():

x_fake = mode.from_tensor(x)
x_fake.resize_([1, 1, 4])
del x_fake

x_fake_2 = mode.from_tensor(x)
print(x_fake_2.shape)
```

Do we hit the cached `[1, 1, 4]` shape ?

Reply:
Yup, we create a new fake mode